tuning/autotune.py

```diff
@@ -962,45 +986,19 @@ def compile_unet_candidates(
 
 
 def sort_candidates_by_first_benchmark_times(
-    candidate_indexes: list[int], candidate_trackers: CandidateTracker
+    candidate_indexes: list[int], candidate_trackers: list[CandidateTracker]
 ) -> list[int]:
     """Sorts candidate indexes based on their first benchmark times in ascending order"""
+    # Get the first benchmark times, defaulting to a large number if None
     first_benchmark_times = [
-        candidate_trackers[index].first_benchmark_time for index in candidate_indexes
+        candidate_trackers[index].first_benchmark_time or float("inf")
+        for index in candidate_indexes
     ]
     combined = list(zip(candidate_indexes, first_benchmark_times))
     combined_sorted = sorted(combined, key=lambda x: x[1])
     sorted_indexes, _ = zip(*combined_sorted)
-    sorted_indexes = list(sorted_indexes)
-    return sorted_indexes
 
-
-def group_benchmark_results_by_device_id(
-    benchmark_results: list[TaskResult],
-) -> list[list[TaskResult]]:
-    """
-    Groups benchmark results by device ID.
-
-    e.g.
-    [TaskResult(res1, device_1), TaskResult(res2, device_2), TaskResult(res3, device_1)]
-    ----->
-    [ [TaskResult(res1, device_1), TaskResult(res3, device_1)], [TaskResult(res2, device_2)] ]
-    """
-    grouped_results = [
-        list(group)
-        for _, group in groupby(benchmark_results, key=lambda br: br.device_id)
-    ]
-    grouped_results: dict[int, list[TaskResult]] = {}
-    for result in benchmark_results:
-        if result.device_id not in grouped_results:
-            grouped_results[result.device_id] = []
-        grouped_results[result.device_id].append(result)
-
-    grouped_benchmark_results = [
-        grouped_results[device_id] for device_id in sorted(grouped_results)
-    ]
-
-    return grouped_benchmark_results
```

Comment:
removed duplicate `group_benchmark_results_by_device_id()` (we have one same copy below), which was probably caused by some rebase errors